src/mlpack/core/data/save_image.hpp

```diff
@@ -1,5 +1,5 @@
 /**
- * @file core/data/save_image.cpp
+ * @file core/data/save_image.hpp
```

Comment:
I think we should rename this file as `save_image_impl.hpp` since it contains implementations. :+1:

Reply:
This comment still applies---do you think we can handle it before merge?